pkg/apis/core/helper/helpers.go

```diff
@@ -167,3 +167,63 @@ func FindWorkerByName(workers []core.Worker, name string) *core.Worker {
 	}
 	return nil
 }
+
+// GetRemovedVersions finds versions that have been removed in the old compared to the new version slice.
+// returns a map associating the index in the old version slice with the version
+func GetRemovedVersions(old []core.ExpirableVersion, new []core.ExpirableVersion) map[int]core.ExpirableVersion {
+	newVersions := make(map[string]struct{}, len(old))
+	for _, x := range new {
+		newVersions[x.Version] = struct{}{}
+	}
+	diff := make(map[int]core.ExpirableVersion, len(old))
+	for index, x := range old {
+		if _, found := newVersions[x.Version]; !found {
+			diff[index] = x
+		}
+	}
+	return diff
+}
+
+// GetAddedVersions finds versions that have been added in the new compared to the new version slice.
+// returns a map associating the index in the new version slice with the version
+func GetAddedVersions(old []core.ExpirableVersion, new []core.ExpirableVersion) map[int]core.ExpirableVersion {
```

Comment:
This function looks almost identical as the above function. Can you move the logic out and just parametrize it accordingly to prevent duplication?

Reply:
Also, as far as I can see the index is only used for the field path error. Can you enhance the data structure so that you can reuse in the admission plugin in #2106?

Reply:
Also returning a sets.Set  containing the versions to be able to reuse them in the admission plugin.